swap_meet/vendor.py

```diff
@@ -1,2 +1,129 @@
+from .item import Item 
 class Vendor:
-    pass
+    #Wave 01
+    def __init__(self,inventory=None): 
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self,item):
+        '''
+        input: one item
+        - adds item to the inventory
+        output: returns item added 
+        '''
+        self.inventory.append(item)
+        return item
+    def remove(self,item):
+        '''
+        input: one item
+        - if there is a mathcing item, removes item from inventory  
+        output: returns item removed if item is valid(matching item) 
+                else: returns False
+        '''
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False 
+
+    #Wave 02
+    def get_by_category(self,category):
+        '''
+        input: a string representing a category
+        output: returns list of items in the 
+        inventory(list) with that category
+        '''
+        result_list = []
+        for item in self.inventory:
+            if category == item.category:
+                result_list.append(item)
+        return result_list
+        
+        
+
+    #Wave 03
+    def swap_items(self,other_vendor, my_item,their_item):
+        '''
+        input: instance of (other) Vendor,
+            instance of Item (this vendor plans to give), 
+            instance of Item (other vendor plans to give)
+        - if both items present in correspeonding Vendor inventory list
+            removes item from corresponding inventory list and adds to 
+            inventory list of oppisite vendor.
+        output: returns True: if items in list and swapped
+                else: returns False
+        '''
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.remove(my_item)
+            self.add(their_item) 
+            other_vendor.remove(their_item)
+            other_vendor.add(my_item)
+            return True
+        else:
+            return False
+            
+
+
+        
+
+    #Wave 04
+    def swap_first_item(self,other_vendor):
+        '''
+        input: instance of (other) Vendor
+        - if both list are valid (not empty), switches first item of 
+            each other vendors and this vendor 
+            inventory list with each other. 
+        output: returns False: if either inventory empty
+                else: returns True 
+        '''
+        if  bool(self.inventory) == False or  bool(other_vendor.inventory) == False:
+            return False
+        else: 
```

Comment:
You already have a function that will do this work for you! `swap_items()` can be called here and then you would pass in the first item of the vender and the other vendor as arguments
`return self.swap_items(other, self.inventory[0], other.inventory[0])`